pkg/helper/machinepools/helpers.go

```diff
@@ -159,6 +164,24 @@ func ParseTaints(taints string) ([]*cmv1.TaintBuilder, error) {
 	return taintBuilders, nil
 }
 
+func validateMachinePoolTaintEffect(taint string) error {
+	var allowedTaintEffects = []string{
+		"NoSchedule",
+		"NoExecute",
+		"PreferNoSchedule",
+	}
+	parts := strings.Split(taint, ":")
+	if len(parts) != 2 {
+		return fmt.Errorf("Invalid taint format: '%s'. Expected format is '<key>=<value>:<effect>'", taint)
+	}
+	effect := parts[1]
+	if effect != "" && !slices.Contains(allowedTaintEffects, effect) {
```

Comment:
@den-rgb Is an empty string a valid effect? This currently allows `""` to be valid. My expectation is that we can just simplify this check here to only `!slices.Contains(allowedTaintEffects, effect)` which would additionally prevent the user from passing an empty string, because it's not in the slice.